model/docs/time.qmd

```diff
@@ -0,0 +1,38 @@
+---
+title: Time handling in pyrenew
+format: gfm
+engine: jupyter
+---
+
+The fundamental time unit in `pyrenew` is "days." `pyrenew` deals with time having `RandomVariable`s carrying information about (i) the time unit in terms of days `t_unit` and (ii) the starting time, `t_start`, measured in days since the first observation. Time series data is anchored to the first observation in the model, which is considered day 0.
```

Comment:
I don't think we should specify that fundamental unit = days. I think we should specify that any given model has a fundamental unit, which could represent any fixed duration duration (e.g. a day).

> Time series data is anchored to the first observation in the model, which is considered day 0.

I don't think we should enforce this.

>  the starting time, `t_start`, measured in days

Is it perhaps clearer to measure `t_start` in the `RandomVariable`'s units, not in fundamental units?

```suggestion
Every `pyrenew` model has a _fundamental time unit_ in which all durations are measured. This should be a unit of approximately fixed length (e.g. "months" would not make for a good fundamental time unit). For many models of interest, this will be days, and we will proceed with this tutorial treating days as our fundamental unit.

 `pyrenew` deals with time having `RandomVariable`s carry information about (i) their own time unit expressed relative to the fundamental unit (`t_unit`) and (ii) the starting time, `t_start`, measured relative to `t = 0` in model time in units of the `RandomVariable`'s time unit. 
```

Reply:
Can we add somehting about what the fundamental time unit is used for?

Reply:
> I don't think we should specify that fundamental unit = days. I think we should specify that any given model has a fundamental unit, which could represent any fixed duration duration (e.g. a day).

Can we add something about what the fundamental time unit is used for?

> > Time series data is anchored to the first observation in the model, which is considered day 0.
> 
> I don't think we should enforce this.
> 
> > the starting time, `t_start`, measured in days
> 
> Is it perhaps clearer to measure `t_start` in the `RandomVariable`'s units, not in fundamental units?

I don't think so. Consider when the fundamental unit is "days." It seems easier to think of/specify "a weekly process starting on day 10" than "a weekly process starting on week 1.42."

Reply:
> I don't think so. Consider when the fundamental unit is "days." It seems easier to think of/specify "a weekly process starting on day 10" than "a weekly process starting on week 1.42."

Good point. And when it _is_ a round number of non-fundamental units the conversion is also easy.

Reply:
> Can we add something about what the fundamental time unit is used for?

Revised suggestion 

```suggestion
Every `pyrenew` model has a _fundamental discrete time unit_. All other time-aware arrays should be expressible as integer fractions or multiples of this fundamental unit.

The fundamental time unit should represent a period of fixed (or approximately fixed) duration. That is, "days" could make for a good fundamental time unit; "months" would not, since different months represent different absolute lengths of time. 

For many infectious disease renewal models of interest, the fundamental time unit will be days, and we will proceed with this tutorial treating days as our fundamental unit.

 `pyrenew` deals with time having `RandomVariable`s carry information about (i) their own time unit expressed relative to the fundamental unit (`t_unit`) and (ii) the starting time, `t_start`, measured relative to `t = 0` in model time in units of the `RandomVariable`'s time unit. 
```

Reply:
My main remaining question is whether we should permit (or anticipate permitting) time units that are integer fractions (as well as integer multiples) of the fundamental time unit. I lean no, as I think supporting this safely might be too much headache for too little gain.

Reply:
Agreed. I think "1 fundamental time unit" should be the smallest time delta allowed in the model.

Reply:
I'll add something about delta time units as part of the design principles to be considered during implementation.

Reply:
Now that I read it, it sounds good as we are not putting any constraints on what the fundamental time unit should be.